src/java/htsjdk/samtools/metrics/MetricsFile.java

```diff
@@ -84,6 +84,13 @@ public void addAllMetrics(final Iterable<BEAN> beanz) {
         else return null;
     }
 
+    /** Returns the histogram contained in the metrics file if any. Adds capability of retrieving multiple histograms
+     *  from metrics file. */
+    public Histogram<HKEY> getHistogram(final int index) {
+        try { return this.histograms.get(index); }
+        catch (final ArrayIndexOutOfBoundsException excep) { throw new SAMException("Index is out of bounds: " + index); }
```

Comment:
Also, in this case the `SAMException` isn't really providing any more information (and it's hiding some...).  I would probably either just not catch this exception and let the `ArrayIndexOutOfBoundsException` propagate up, or if you think a SAMException is more appropriate, I would give it a bit more context in the message.  `"There is no histogram with index " +index+ "."